main_supcon.py

```diff
@@ -305,15 +316,16 @@ def main():
         # tensorboard logger
         logger.log_value('loss', loss, epoch)
         logger.log_value('learning_rate', optimizer.param_groups[0]['lr'], epoch)
+        print(f'loss: {loss} epoch {epoch}')
 
         if epoch % opt.save_freq == 0:
             save_file = os.path.join(
-                opt.save_folder, 'ckpt_epoch_{epoch}.pth'.format(epoch=epoch))
+                opt.save_folder, f'ckpt_{opt.data_folder.split("/")[-2]}_epoch_{epoch}.pth')
```

Comment:
maybe make this an argument name you can pass in ? this won't work for me right now because I don't put full data path